### DISCUSSION TITLE
授权报错 重定向问题

### DISCUSSION THREAD

**Discussion by zhangiscode** [w7corp/easywechat#2635] [General]
laravel 版本6.20
overtrue/laravel-wechat  版本5.0.11
在执行重定向时，使用：
redirect()->away(
                $officialAccount->oauth->scopes($scope)
                                       ->redirect($this->getRedirectUrl($request, $enforceHttps))
            );
            报错：Header may not contain more than a single header, new line detected
      换成： return $officialAccount->oauth->scopes($scope)->redirect($request->fullUrl()); 就可以了！麻烦能解答一下吗？

**Comment by overtrue** [comment]
```
$officialAccount->oauth->scopes($scope)->redirect($request->fullUrl())
```
上面这行的返回值本身就是一个 RedirectResponse 了，不需要再用框架的跳转函数包一层。

**Reply by zhangiscode** [reply]
但是包里面使用的是这个
![image](https://user-images.githubusercontent.com/45380983/207526296-b0b94413-cc60-4d5d-9106-2fc9253f7e83.png)

**Reply by overtrue** [reply]
```bash
composer info overtrue/wechat
```

看看输出啥

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207527971-26097219-f47c-4879-85ec-bd73f9dc47c4.png)

**Reply by zhangiscode** [reply]
5.0.9 版本 用的是$officialAccount->oauth->scopes($scope)->redirect($request->fullUrl()); 他是有效的

**Comment by overtrue** [comment]
1. 检查一下你的配置文件跳转 URL 是否有换行等其他字符
2. 看看中间件里 `$this->getRedirectUrl($request, $enforceHttps)` 得到的是啥，是否包含换行

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207529643-39b3e8dc-1224-4d33-b212-ea08a64f52b0.png)
![image](https://user-images.githubusercontent.com/45380983/207529679-d05b656f-fa4c-4d98-9cf6-43bc0ae2361d.png)
这个检查过了 没有换行

**Reply by overtrue** [reply]
那就 `dd(redirect()->away(
$officialAccount->oauth->scopes($scope)
->redirect($this->getRedirectUrl($request, $enforceHttps))
))`;看看

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207530463-6799b265-0188-422c-8169-1cc87e032578.png)
这里面参数有点多 不知道到底拿的哪个跳转的

**Reply by overtrue** [reply]
看起来没问题哦，你装了什么其他调试工具之类的没

**Reply by zhangiscode** [reply]
就一个微信开发者工具

**Reply by overtrue** [reply]
我说的是 composer.json 里，贴一下 composer.json ？

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207531239-f7bb3fbb-df82-4b09-aa64-a13292ea8069.png)

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207531708-e3550db1-68e1-49d2-94db-d3948b5033e6.png)

**Reply by overtrue** [reply]
require-dev 也贴一下

**Reply by overtrue** [reply]
把报错消息完整贴一下，看看是哪里报的

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207532306-119832fc-7150-47c3-895e-7c5a2561bac8.png)

**Reply by overtrue** [reply]
进到trace 1 Response 346 行，把输入打一下

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207533288-64a61066-69da-4ff3-a5bb-f9b73508a0d4.png)
![image](https://user-images.githubusercontent.com/45380983/207533305-a68524a4-9202-49d3-b72a-d7ad3536c2e1.png)

**Reply by overtrue** [reply]
![image](https://user-images.githubusercontent.com/1472352/207533745-66fb0bf6-bfa5-4d05-b55f-49dda3365ed3.png)

把这里改一下，上面那个 dd 不需要，把这里的 header() 调用改成这样子：

```php
try {
   header($name.': '.$value, $replace, $this->statusCode);
} catch(\Throwable $e) {
   dd($name, $value);
} 
```

**Reply by zhangiscode** [reply]
![image](https://user-images.githubusercontent.com/45380983/207534288-64666d00-f836-4cf8-a31b-61ca77f2ea4f.png)

**Reply by overtrue** [reply]
我知道了，稍等

**Reply by zhangiscode** [reply]
ok

**Reply by overtrue** [reply]
```bash
composer require overtrue/laravel-wechat:^5.1.0
```

**Reply by zhangiscode** [reply]
ok

**Reply by zhangiscode** [reply]
可以了  麻烦解答一下吗？

**Reply by overtrue** [reply]
因为 5.1.0 修复了 5.0.12 的一个bug,而你的版本恰好是有bug的版本：https://github.com/overtrue/laravel-wechat/compare/5.0.12...5.1.0#diff-76c7f4416a57d7988aaf5d8aaa1f2dc94d7dfb7f309a1c0dd5f67ed065d2ba5eR73

**Reply by zhangiscode** [reply]
好吧  谢谢了

**Reply by overtrue** [reply]
不客气
